databricks_cli/sdk/service.py

```diff
@@ -519,25 +523,45 @@ def list_test(self, path, headers=None):
             _data['path'] = path
         return self.client.perform_query('GET', '/dbfs-testing/list', data=_data, headers=headers)
 
-    def put(self, path, contents=None, overwrite=None, headers=None):
+    def put(self, path, contents=None, overwrite=None, headers=None, src_path=None):
         _data = {}
+        _files = None
         if path is not None:
             _data['path'] = path
         if contents is not None:
-            _data['contents'] = contents
+            # Because terminal might add trailing newlines, they need to be encoded properly.
+            encoded_contents = base64.b64encode(contents.encode('utf-8'))
+            _data['contents'] = encoded_contents.decode("utf-8")
         if overwrite is not None:
             _data['overwrite'] = overwrite
-        return self.client.perform_query('POST', '/dbfs/put', data=_data, headers=headers)
-
-    def put_test(self, path, contents=None, overwrite=None, headers=None):
-        _data = {}
+        if src_path is not None:
+            # @self.client sets Content-Type 'text/json' by default.
+            # For multipart/form-data POST Content-Type should be set automatically
+            # to decode 'Boundary' parameter.
+            headers = {'Content-Type': None}
```

Comment:
This doesn't seem right, The content Type is expected to be something of this format 
`Content-Type: multipart/form-data; boundary=something` 
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type

See how we check whether a request is a multipart upload or not // https://livegrep.dev.databricks.com/view/databricks/universe/daemon/data/daemon/src/main/scala/com/databricks/backend/daemon/data/server/meta/DbfsFileUploadDownloadBackend.scala#L305

Reply:
I have checked this quite a lot. If we set the `content-type` manually, requests library forces programmer to define other required fields such as `boundary`. If the `content-type` is not set (or `None`), requests library automatically fills them. If a `files` parameter is passed to the call, it will automatically generate `Content-Type: multipart/form-data; boundary=something`.

A lot of answers on stack I checked were against setting 'content-type' in this case.